specification/cdn/resource-manager/Microsoft.Cdn/preview/2019-06-15-preview/cdnwebapplicationfirewall.json

```diff
@@ -0,0 +1,919 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2019-06-15-preview",
+    "title": "Azure CDN WebApplicationFirewallManagement",
+    "description": "APIs to manage web application firewall rules for Azure CDN"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cdn/CdnWebApplicationFirewallPolicies": {
+      "get": {
+        "tags": [
+          "WebApplicationFirewallPolicies"
+        ],
+        "description": "Lists all of the protection policies within a resource group.",
+        "operationId": "Policies_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/CdnWebApplicationFirewallPolicyList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./cdn.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List Policies in a Resource Group": {
+            "$ref": "./examples/WafListPolicies.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cdn/CdnWebApplicationFirewallPolicies/{policyName}": {
```

Comment:
What's the relationship between a cdnWebApplicationFirewallPolicies resource type and a [Microsoft.Network/webApplicationFirewallPolicies ](https://github.com/Azure/azure-rest-api-specs/blob/7816a440b356998df165e0dd84aa368794c22028/specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/webapplicationfirewall.json )resource ?  Is there an opportunity to use an extension resource, such as {resource URI}/providers/Microsoft.Network/webApplicationFirewallsPolicies/{resource name} in lieu of a new duplicate resource type? If not, does the schema deviate from the .Network variant?

Reply:
This is not a duplicate, it is for Azure Cdn product (which works with multiple third party providers like Akamai and Verizon) and has a different schema and is expected to evolve seperately.